### ISSUE TITLE
Waterfox 56.2.3 (UA … rv:56.0; Waterfox) Gecko/20100101 Firefox/56.2.3) outdated according to the Outdated Browser service

### ISSUE THREAD

**Issue by grahamperrin** [burocratik/outdated-browser#294]
Hi

Please be aware that Waterfox 56.2.0  ([tagged in GitHub](https://github.com/MrAlex94/Waterfox/tree/56.2.0)) is a recent release, not yet outdated. 

Given what's below, I don't expect the Outdated Browser service to grant an exception. 

Any advice that you might like to share will be gratefully appreciated. 

## Essentially

For Waterfox 56.2.0, https://udger.com/resources/online-parser reports, in my (edge) case: 

`Mozilla/5.0 (X11; FreeBSD amd64; rv:56.0) Gecko/20100101 Firefox/56.0`

– no hint that it's anything other than an outdated Mozilla Firefox. 

## Background

A few brief points.

### Recent months

1. Waterfox 56.0 was based on Firefox 56.0.2 whilst Firefox 56.0.2 was supported by Mozilla
2. introduction of a default Waterfox-specific user agent string proved to be an issue for a significant number of sites – maybe not a problem for 'power users', but enough of an issue for a subsequent release of Waterfox to (again) default to a string that is primarily associated with Mozilla Firefox
3. https://github.com/browser-update/browser-update/issues/397 (2018-05-23) understandably notes that given the current default UA string of Waterfox, Waterfox **cannot be distinguished** from outdated Firefox
4. re: <https://github.com/MrAlex94/Waterfox/issues/178#issuecomment-391486980> (2018-05-23) under _User-Agent: Waterfox · Issue `#178` · MrAlex94/Waterfox_ I began experimenting with **string A** below. 

### Looking ahead

From [The Waterfox Blog | Waterfox, Its Legacy and Looking to the Future](https://blog.waterfoxproject.org/waterfox-its-legacy-and-looking-to-the-future) (2018-04-28): 

> … The future variant will be:
> 
> - Based on ESR releases …

Some of the technical flesh to that bone is, if you like, at <https://github.com/MrAlex94/Waterfox/issues/267#issuecomment-376351366> (2018-03-27) under _Waterfox Quantum? · Issue `#267` · MrAlex94/Waterfox_

## Experimental string A

`Mozilla/5.0 (X11; FreeBSD amd64; rv:56.2.0) Gecko/20100101 Firefox/56.2.0`

Truly there will never be a Firefox 56.2.0. 

The **distinction** might allow services such as http://outdatedbrowser.com/ and Browser-Update.org to be enhanced; maybe to associate `Firefox/56.2.0` with Waterfox (not with Firefox) and recognise Waterfox 56.2.0 as current. 

(Note to self: experiments with this string A lapsed when other things took priority. Resuming this morning.)

## Experimental string B

`Mozilla/5.0 (X11; FreeBSD amd64; rv:60.0) Gecko/20100101 Firefox/60.0`

Of course that's _not_ experimental where the browser is truly Firefox. However if I choose to use that string for Waterfox 56.2.0 – and this is not an invitation to others to follow suit, at this time – it'll be sane for me to treat it as experimental. 

## My own environment – edge case

```text
$ pkg info waterfox | grep Version
Version        : 56.2.0.31
$ date ; uname -v
Thu 31 May 2018 08:08:04 BST
FreeBSD 12.0-CURRENT #3 r334382: Thu May 31 01:37:12 BST 2018     root@momh167-gjp4-hpelitebook8570p-freebsd:/usr/obj/usr/src/amd64.amd64/sys/GENERIC 
```

That's exceptional, edge case, FreeBSD. The majority of Waterfox users use platforms that are Tier-1 for Mozilla Firefox support purposes.

**Comment by grahamperrin** [comment]
Meta, tracking: [(LIST) Waterfox issues yet unresolved. · Issue #538 · MrAlex94/Waterfox](https://github.com/MrAlex94/Waterfox/issues/538)

**Comment by grahamperrin** [comment]
Re: the closing comment in browser-update/browser-update issue 397

In lieu of experimental string 'A', 

`Mozilla/5.0 (X11; FreeBSD amd64; rv:56.2.0) Gecko/20100101 Firefox/56.2.0`

I'm now using an experimental string 'A-01', 

`Mozilla/5.0 (X11; FreeBSD amd64; rv:56.2) Gecko/20100101 Firefox/56.2`

**Comment by grahamperrin** [comment]
From the opening post: 

> no hint that it's anything other than an outdated Mozilla Firefox.

### tl;dr

**Strings for Waterfox are now distinct from strings for Firefox** :+1: 

The three strings that are primarily associated with Waterfox 56.2.3 ([released yesterday](https://blog.waterfoxproject.org/waterfox-56.2.3-release-download)):

- Mozilla/5.0 (Windows NT x.y; Win64; x64; rv:56.0; Waterfox) Gecko/20100101 Firefox/56.2.3
- Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:56.0; Waterfox) Gecko/20100101 Firefox/56.2.3
- Mozilla/5.0 (X11; Linux x86_64; rv:56.0; Waterfox) Gecko/20100101 Firefox/56.2.3

Please, are these suitable for whitelisting?

**Comment by SuperOP535** [comment]
https://github.com/burocratik/outdated-browser/wiki/FAQ#-went-to-the-website-outdatedbrowsercom-and-i-get-a-message-saying-that-my-browser-is-outdated-when-its-not

**Comment by grahamperrin** [comment]
@SuperOP535 I did recently read the FAQ (thanks) but didn't want to repeat myself, https://github.com/burocratik/outdated-browser/issues/296#issuecomment-422267890 asks: 

> Is there a URL I can visit to more positively test?
